app/main.py

```diff
@@ -1,3 +1,56 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int):
+        self.km = km
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other):
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
```

Comment:
In the `__add__` method, ensure that `other` is a number (int or float) when it's not a `Distance` instance. This will prevent potential errors if `other` is of an unexpected type.